caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -750,3 +654,172 @@ def save(self, model_path: str, *args, **kwargs):
+        else:
+            okay_to_truncate = False
+            max_length = max_tokens
+
+        if isinstance(texts[0], str):
+            to_tokenize = [texts]
+        else:
+            assert 0
+
+        to_tokenize = [[str(s).strip() for s in col] for col in to_tokenize]
+        tokenized = self.tokenizer(
+            *to_tokenize,
+            return_attention_mask=True,
+            return_token_type_ids=False,
+            return_overflowing_tokens=True,
+            return_offsets_mapping=True,
+            return_length=True,
+            return_tensors="pt",
+            truncation=True,
+            padding=True,
+            max_length=max_length,
+        )
+
+        texts_map = tokenized["overflow_to_sample_mapping"]
+
+        for text_number, text in enumerate(texts):
+            # positions: the positions (in lengths and offsets arrays) that belong to this text
+            positions = [
+                position
+                for position, sample_number in enumerate(texts_map)
+                if sample_number == text_number
+            ]
+            lengths = [tokenized["length"][pos] for pos in positions]
+
+            was_truncated = len(lengths) > 1  # multiple lengths when truncated
+
+            if not okay_to_truncate and was_truncated:
+                # On truncation errors, we're asked to provide the actual tokens vs limit in the
+                # error message (like some models typically do). In order to calculate the tokens
+                # we will re-tokenize without padding to get that length for the error message.
+                re_tokenized = self.tokenizer(
+                    text,
+                    return_attention_mask=False,
+                    return_token_type_ids=False,
+                    return_overflowing_tokens=True,
+                    return_offsets_mapping=False,
+                    return_length=True,
+                    truncation=True,
+                    max_length=max_length,
+                )
+                re_lengths = [re_tokenized["length"][pos] for pos in positions]
+                # Raise error. We don't allow silent truncation in this case.
+                tokens = (
+                    sum(re_lengths) - 2
+                )  # add up total tokens for error message (-2 begin/end)
+                error.log_raise(
+                    "<NLP08391926E>",
+                    ValueError(
+                        f"Token sequence length is longer than the specified "
+                        f"maximum sequence length for this model ({tokens} > {max_tokens})."
+                    ),
+                )
+
+        return tokenized
+
+    def encode(
+        self,
+        sentences: Union[str, List[str]],
+        batch_size: int = 32,
+        device: str = None,
+        convert_to_numpy: bool = True,
+        convert_to_tensor: bool = False,
+        truncate_input_tokens: Optional[int] = 0,
+    ) -> np.ndarray:
```

Comment:
I think a docstring here could be helpful especially considering it's a non-private method